libraries/OPAMP/src/OPAMP.h

```diff
@@ -0,0 +1,52 @@
+#ifndef _OPAMP_H_
+#define _OPAMP_H_
+
+#include <stdint.h>
+
+/* Available speed modes */
+/* Note: No middle speed mode on the Uno R4 boards */
+enum OpampSpeedMode {
+    OPAMP_SPEED_LOWSPEED = 0,
+    OPAMP_SPEED_HIGHSPEED = 1,
+};
+
+/* The supported boards have 4 OPAMP channels, however, only channel 0 is accessible. */
+/* All other channels are connected to the LED matrix or not exposed. */
+#define ARDUINO_DEFAULT_OPAMP_CHANNEL 0
+
+/**
+ * Pin Mapping for OPAMP
+ * Uno R4 (Minima, WiFi):
+ * ~Channel 0~
+ *  Plus: Analog A1 (Renesas P0.00)
+ *  Minus: Analog A2 (Renesas P0.01) 
+ *  Output: Analog A3 (Renesas P0.02)
+ * ~Channel 1~ (Inaccessible)
+ *  +: P0.13, -: Renesas P0.12, Out: Renesas P0.03
+ * ~Channel 2~ (Inaccessible)
+ *  +: P0.11, -: Renesas P0.10, Out: Renesas P0.04
+ * ~Channel 3~ (Inaccessible)
+ *  +: P0.05, -: Renesas P0.06, Out: Renesas P0.07
+*/
+class OpampClass {
+public:
+    /* startup the OPAMP on channel 0 */
+    bool begin(OpampSpeedMode speed = OPAMP_SPEED_HIGHSPEED);
+    /* startup the OPAMP with arbitrary channel mask */
+    bool begin(uint8_t channel_mask, OpampSpeedMode speed = OPAMP_SPEED_HIGHSPEED);
+    /* stop all OPAMP channels */
+    void end();
+    /* stop specific OPAMP channel(s) */
+    void end(uint8_t channel_mask);

```

Comment:
I am not sure whether this would enable any compiler optimization for better codegen at all to mark integers, that are copied by-value anyways, as `const`. Done so regardless in 6dd13b199472122a9de9c4ce5033c2a460ccf190. Should this be done on the enums too?

Reply:
Const-correctness is not about compiler optimisation. It's about safe code. I do not compromise on that :grin: at all :grin: :grin: :grin: :angel: .